json/decode.go

```diff
@@ -617,11 +618,19 @@ func (d *decodeState) object(v reflect.Value) {
 		start := d.off - 1
 		op = d.scanWhile(scanContinue)
 		item := d.data[start : d.off-1]
-		key, ok := unquoteBytes(item)
+		key, ok := unquote(item)
```

Comment:
I'm not sure what's going on here.  Why are switching from bytes to string here?